eth2/types/src/chain_spec.rs

```diff
@@ -315,3 +316,366 @@ mod tests {
+            domain_deposit: self.domain_deposit,
+            domain_voluntary_exit: self.domain_voluntary_exit,
+            boot_nodes: chain_spec.boot_nodes.clone(),
+            genesis_fork: chain_spec.genesis_fork.clone(),
+            ..*chain_spec
+        })
+    }
+}
+
+#[cfg(test)]
+mod yaml_tests {
+    use super::*;
+    use std::fs::OpenOptions;
+    use tempfile::NamedTempFile;
+
+    #[test]
+    fn minimal_round_trip() {
+        // create temp file
+        let tmp_file = NamedTempFile::new().expect("failed to create temp file");
+        let writer = OpenOptions::new()
+            .read(false)
+            .write(true)
+            .open(tmp_file.as_ref())
+            .expect("error opening file");
+        let minimal_spec = ChainSpec::minimal();
+
+        let yamlconfig = YamlConfig::from_spec::<MinimalEthSpec>(&minimal_spec);
+        // write fresh minimal config to file
+        serde_yaml::to_writer(writer, &yamlconfig).expect("failed to write or serialize");
+
+        let reader = OpenOptions::new()
+            .read(true)
+            .write(false)
+            .open(tmp_file.as_ref())
+            .expect("error while opening the file");
+        // deserialize minimal config from file
+        let from: YamlConfig = serde_yaml::from_reader(reader).expect("error while deserializing");
+        assert_eq!(from, yamlconfig);
+    }
+
+    #[test]
+    fn mainnet_round_trip() {
+        let tmp_file = NamedTempFile::new().expect("failed to create temp file");
+        let writer = OpenOptions::new()
+            .read(false)
+            .write(true)
+            .open(tmp_file.as_ref())
+            .expect("error opening file");
+        let mainnet_spec = ChainSpec::mainnet();
+        let yamlconfig = YamlConfig::from_spec::<MainnetEthSpec>(&mainnet_spec);
+        serde_yaml::to_writer(writer, &yamlconfig).expect("failed to write or serialize");
+
+        let reader = OpenOptions::new()
+            .read(true)
+            .write(false)
+            .open(tmp_file.as_ref())
+            .expect("error while opening the file");
+        let from: YamlConfig = serde_yaml::from_reader(reader).expect("error while deserializing");
+        assert_eq!(from, yamlconfig);
+    }
+
+    #[test]
+    fn apply_to_spec() {
+        let mut spec = ChainSpec::minimal();
+        let yamlconfig = YamlConfig::from_spec::<MinimalEthSpec>(&spec);
+
+        // modifying the original spec
+        spec.deposit_contract_tree_depth += 1;
+        // Applying a yaml config with incorrect EthSpec should fail
+        let res = yamlconfig.apply_to_chain_spec::<MainnetEthSpec>(&spec);
+        assert_eq!(res, None);
+
+        // Applying a yaml config with correct EthSpec should NOT fail
+        let new_spec = yamlconfig
+            .apply_to_chain_spec::<MinimalEthSpec>(&spec)
+            .expect("should have applied spec");
+        assert_eq!(new_spec, ChainSpec::minimal());
+    }
+
+    // Stub for testing deserializing
```

Comment:
How come these stubbed out tests are included?

Reply:
I thought they would be useful for testing for later changes in specs, but I'll remove them and keep them locally :D